### ISSUE TITLE
Compatibility issues with PHP8

### ISSUE THREAD

**Issue by timdreier** [mmunz/c1_adaptive_images#19]
Hi,
just found some compatibility issues with PHP8. My be an misconfiguration, but since it worked with PHP7 it should be fixed in my opinion! Got the following exception...

`Exception: PHP Warning: Undefined array key "media"`

... with the following code:

```
<ai:picture image="{files.0}"
    class="img-responsive-full lazyload heroheader__image"
    width="{dimensions.width}"
    height="{dimensions.height}"
    alt="{files.0.alternative}"
    title="{files.0.title}"
    sources="{
        'mobile': {
            'srcsetWidths': '320,1024,1600',
            'media': '(max-width: 767px)'
        },
        'desktop': {
            'srcsetWidths': '320,1024,1600',
            'media': '(min-width: 768px)'
        }
    }"
    cropVariant="desktop"
    srcsetWidths="320,1024,1600"
    placeholderInline="1"
    placeholderWidth="64"
    lazy="1"
    debug="0"
    ratiobox="1"
    jsdebug="0"
    sizes="100vw"
/>
```

For now i temporarily fixed it by changing the following lines to:

PictureViewHelper:102 `$mq[$key] = $config['media'] ?? NULL;`

Add after ImageUtility:238 `unset($localProcessingConfiguration['height']);`

**Comment by mmunz** [comment]
Hi Tim,

i can't reproduce this with PHP 8.1.3, TYPO3 v11.5.15 and adaptive images v0.4.0.

Can you please try the following:

- remove the desktop array inside sources. This will be unset anyways because it is already used for the default (the image tag)
- probably not the problem, i also tried without, but it is best practice to always have a 'default' crop variant (i would use that instead of 'desktop')
- can you also test with the development branch of c1_adaptive_images?

If that all does not help: can you debug what is inside $this->cropVariants when $mq[$key] = $config['media'] because i can't imagine why media is undefined there.

As a side note: thats 'just' a warning, for some reasons your TYPO3 instance seems to throw errors on warnings. You might also try to change that. But nevertheless, i see more warnings for undefined array keys and will fix those in the development branch.

**Comment by timdreier** [comment]
Hi,
sorry for the late reply, I was on vacation.

There seems no way of surpressing this exception in PHP8. I even asked in the TYPO3 slack, they said that the exception is "intended" with PHP8.

However, the latest dev-Branch seems to work great, thank you!

**Comment by timdreier** [comment]
Hi @mmunz,
will these changes come to a release in the future? Works great, but kind of an ugly solution to use the dev branch in all PHP 8 projects. Thank you for your effort! :)

**Comment by mmunz** [comment]
@timdreier working on it, but will be for v11 and v12 only.

**Comment by mmunz** [comment]
@timdreier finally i fixed all tests and Ci and deployment issues and released v1.0.1.
